components/automate-deployment/pkg/target/hab_cmd.go

```diff
@@ -120,37 +146,76 @@ func (c *habCmd) IsInstalled(pkg habpkg.VersionedPackage) (bool, error) {
 }
 
 // BinlinkPackage binlinks an executable from a Habitat package
-func (c *habCmd) BinlinkPackage(pkg habpkg.VersionedPackage, exe string) (string, error) {
+func (c *habCmd) BinlinkPackage(ctx context.Context, pkg habpkg.VersionedPackage, exe string) (string, error) {
 	args := command.Args("pkg", "binlink", "--force", habpkg.Ident(pkg), exe)
-	return c.executor.CombinedOutput("hab", append(standardHabOptions(), args)...)
+	ctx, cancel := context.WithTimeout(ctx, HabTimeoutDefault)
+	defer cancel()
+
+	cmdOpts := append(standardHabOptions(), command.Context(ctx), args)
+
+	return c.executor.CombinedOutput("hab", cmdOpts...)
 }
 
-func (c *habCmd) LoadService(svc habpkg.VersionedPackage, opts ...LoadOption) (string, error) {
+func (c *habCmd) LoadService(ctx context.Context, svc habpkg.VersionedPackage, opts ...LoadOption) (string, error) {
 	args := []string{"svc", "load", "--force", habpkg.Ident(svc), "--strategy", "none"}
 	for _, o := range opts {
 		args = o(args)
 	}
-	return c.executor.CombinedOutput("hab", append(standardHabOptions(), command.Args(args...))...)
+
+	ctx, cancel := context.WithTimeout(ctx, HabTimeoutDefault)
+	defer cancel()
+
+	cmdOpts := append(standardHabOptions(), command.Context(ctx), command.Args(args...))
+
+	return c.executor.CombinedOutput("hab", cmdOpts...)
 }
 
-func (c *habCmd) UnloadService(svc habpkg.VersionedPackage) (string, error) {
+func (c *habCmd) UnloadService(ctx context.Context, svc habpkg.VersionedPackage) (string, error) {
 	args := command.Args("svc", "unload", habpkg.ShortIdent(svc))
-	return c.executor.CombinedOutput("hab", append(standardHabOptions(), args)...)
+	ctx, cancel := context.WithTimeout(ctx, HabTimeoutDefault)
+	defer cancel()
+
+	cmdOpts := append(standardHabOptions(), command.Context(ctx), args)
+	output, err := c.executor.CombinedOutput("hab", cmdOpts...)
+
+	// NOTE: hab <= 0.79 returned 0 if the service was already unloaded.
+	// Beginning with 0.80, hab will exit 1 if the service was already unloaded.
+	// As we need to support mixed versions because of upgrades we'll do our
+	// best to handle backwards compatibility of this interface.
+	if err != nil {
+		regex := regexp.MustCompile("not loaded")
```

Comment:
can we pull this out into a global variable so it fails to even start if someone messes it up